pytorch_lightning/trainer/trainer.py

```diff
@@ -1246,8 +1247,8 @@ def run_stage(self) -> None:
         return self._run_stage()
 
     def _run_stage(self):
-        self.strategy.barrier("run-stage")
-        self.strategy.dispatch(self)
+        self.training_type_plugin.barrier("run-stage")
+        self.training_type_plugin.dispatch(self)
         self.__setup_profiler()
```

Comment:
do we need it?

Reply:
this change is from rebase with the master code.